src/box/sql/func.c

```diff
@@ -989,6 +989,14 @@ func_round_double(struct sql_context *ctx, int argc, const struct Mem *argv)
 	assert(mem_is_double(&argv[0]));
 	assert(argc == 1 || mem_is_int(&argv[1]));
 	uint64_t n = (argc == 2 && mem_is_uint(&argv[1])) ? argv[1].u.u : 0;
+	/*
+	 * The smallest positive double value is 2.225E-307, and the value
+	 * before the exponent has a maximum of 15 digits after the decimal
+	 * point. This means that double values cannot have more than 307 + 15
+	 * digits after the decimal point.
+	 */
+	if (n > 322)
+		return mem_copy_as_ephemeral(ctx->pOut, &argv[0]);
 
 	double d = argv[0].u.r;
 	struct Mem *res = ctx->pOut;
```

Comment:
You can also do this now:
```Diff
@@ -1000,11 +1000,8 @@ func_round_double(struct sql_context *ctx, int argc, const struct Mem *argv)
 
 	double d = argv[0].u.r;
 	struct Mem *res = ctx->pOut;
-	if (n != 0) {
-		int precision = MIN(n, INT_MAX);
-		return mem_set_double(res, atof(tt_sprintf(
-			"%.*lf", precision, d)));
-	}
+	if (n != 0)
+		return mem_set_double(res, atof(tt_sprintf("%.*lf", n, d)));
 	/*
```

Reply:
Thank you for the review! Fixed, however I added a bit more:
```Diff
diff --git a/src/box/sql/func.c b/src/box/sql/func.c
index 7ce808d7f..163ae992c 100644
--- a/src/box/sql/func.c
+++ b/src/box/sql/func.c
@@ -1001,9 +1001,8 @@ func_round_double(struct sql_context *ctx, int argc, const struct Mem *argv)
        double d = argv[0].u.r;
        struct Mem *res = ctx->pOut;
        if (n != 0) {
-               int precision = MIN(n, INT_MAX);
-               return mem_set_double(res, atof(tt_sprintf(
-                       "%.*lf", precision, d)));
+               d = atof(tt_sprintf("%.*lf", (int)n, d));
+               return mem_set_double(res, d);
        }
        /*
         * DOUBLE values greater than 2^53 or less than -2^53 have no digits
```

I did this because `%.*lf` throws an error if `n` is not `int`. When I tried to cast, the line got too long.